server/src/main/java/io/crate/metadata/table/ConstraintInfo.java

```diff
@@ -89,19 +89,10 @@ private static List<Short> getConstraintColumnIndices(RelationInfo relationInfo,
                     }
                 }
             }
-        } else if (constraintType == Type.CHECK) {
-            var checks = relationInfo.checkConstraints();
-            for (int i = 0; i < relationInfo.checkConstraints().size(); i++) {
-                var iterator = relationInfo.columns().iterator();
-                while (iterator.hasNext()) {
-                    var ref = iterator.next();
-                    if (ref.column().name().equals(checks.get(i).columnName())) {
-                        result.add((short) ref.position());
-                        break;
-                    }
-                }
-            }
         } else {
+            // conkey array is currently populated only for PK.
+            // To make the same for CHECK we need to introduce breaking change and enrich "check_constraints" meta map with column name to resolve position
```

Comment:
actually not breaking if change
`Map<String, String> checkConstraintsMap = Maps.get(metaMap, "check_constraints");`
to `Map<String, Map<String, Object> and check version or instance of in entry value.